### ISSUE TITLE
Sanity checking of parameters/input files

### ISSUE THREAD

**Issue by ppxasjsm** [michellab/Sire#36]
It would be good if we can implement a consistent way of checking input parameters. 
Most Sire apps will have a common set of checks and then app specific set of checks. 
What would be the best way to implement this and where exactly? 
To me a class structure seems useful here, having a SanityCheck Class, that would then have a set of common sanity checks implemented and can then be extended for app specific checks. Would this make sense? What would be the best place to put such checks? 
I would also propose to use something like Run time warnings, and runtime errors to distinguish between severity of badly chosen parameters.

**Comment by chryswoods** [comment]
I agree that it would be useful. The python/Tools/__init__.py has some of this already (admittedly this is a bad place for it, and it should be moved to something like python/Tools/something.py).

This contains the "Parameter" class that is used to describe and document most of the script parameters. I suggest that a ".validate()" member function could be added that could validate the parameter? Parameter is currently a very simple class that records the name of the parameter, default value and the help text. You could extend this to include the datatype of the parameter, a range of valid values, and then code to validate that the user has entered an allowable value.

  Christopher
